src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -440,18 +440,12 @@ private static void InitializeGetEnumerableCache()
 
         internal static bool IsTypeEnumerable(Type type)
         {
+            if (type == null) { return false; }
             GetEnumerableDelegate getEnumerable = GetOrCalculateEnumerable(type);
             return (getEnumerable != LanguagePrimitives.ReturnNullEnumerable);
         }
 
-        /// <summary>
-        /// Retrieves the IEnumerable of obj or null if the language does not consider obj to be IEnumerable
-        /// </summary>
-        /// <param name="obj">
-        /// IEnumerable or IEnumerable-like object
-        /// </param>
-        [SuppressMessage("Microsoft.Naming", "CA1720:IdentifiersShouldNotContainTypeNames", MessageId = "obj", Justification = "Since V1 code is already shipped, excluding this message.")]
-        public static IEnumerable GetEnumerable(object obj)
+        internal static Type GetBaseObjectType(object obj)
```

Comment:
It's recommended to always start from `private`, since currently nothing outside the type depends on it.